BUG: Remove `read_json` datetime deprecation warning

KevsterAmp opened the pull request:
- [x] closes #59511 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

jorisvandenbossche committed:
String dtype: honor mode.string_storage option (and change default to None) (#59488)

* String dtype: honor mode.string_storage option (and change default to None)

* fix test + explicitly test default

* use 'auto' instead of None

tswast committed:
DOC: add pandas-gbq and related db-dtypes packages to ecosystem (#59491)

* DOC: add pandas-gbq and related db-dtypes packages to ecosystem

As of pandas 2.2.0, pandas-gbq functionality is fully separated, so I think now would be a good time to introduce it as part of the "ecosystem".

db-dtypes was built for use from pandas-gbq but should be applicable more generally.

* fix pandas-gbq link

shreyas-dev committed:
DOC: Fix docstrings for Timestamp: unit, utcoffset, utctimetuple (#59496)

* unit, utcoffset, utctimetuple

* Fix E501 line too long cython-lint

abhinav-thimma committed:
DOC: fix PR02 for pandas.tseries.offsets.* (#59485)

* Correcting PR02 for classes in offsets.pyx

* remove remaining PR02 issues in offsets.pyx

* revert for errored classes

* revert dependent classes

---------

Co-authored-by: Abhinav Thimma <athimma2@illinois.edu>

jbrockmendel committed:
REF: remove unused variable and checks (#59463)

* REF: removed unused convert keyword

* REF: remove unhit field checks

* REF: inline _object_ops

jbrockmendel committed:
BUG (string): ArrowEA comparisons with mismatched types (#59505)

* BUG: ArrowEA comparisons with mismatched types

* move whatsnew

* GH ref

kirill-bash committed:
ENH: Allow Iterable[Hashable] in drop_duplicates

awojno-bloomberg committed:
DOC: Docstrings for pandas.Series.str functions (#59506)

Docstrings for pandas.Series.str functions

pandas.Series.str.rpartition
pandas.Series.str.rstrip
pandas.Series.str.strip

jorisvandenbossche committed:
TST (string dtype): clean up construction of expected string arrays (#59481)

jorisvandenbossche committed:
TST (string dtype): fix IO dtype_backend tests for storage of str dtype of columns' Index (#59509)

* TST (string dtype): fix failure in csv test_dtype_backend_string test

* ignore storage of Index str dtype

tan-i-ham committed:
docs: Fix ES01, SA01 code check of `pandas.api.types.is_bool_dtype` (#59507)

* docs: FIx ES01, SA01 code check of `pandas.api.types.is_bool_dtype`

* chore: Remove pandas in See Also

jbrockmendel committed:
REF (string): Move StringArrayNumpySemantics methods to base class (#59514)

* REF (string): Move StringArrayNumpySemantics methods to base class

* mypy fixup

tuhinsharma121 committed:
DOC: fix PR07,SA01 for pandas.MultiIndex.sortlevel (#59521)

tuhinsharma121 committed:
DOC: fix ES01,SA01 for pandas.MultiIndex.names (#59520)

DOC: fix ES01,PR07 for pandas.MultiIndex.names

uditbaliyan committed:
Fix docstrings for pandas.Period.month (#59517)

* Fix docstrings for pandas.Period.month

* Fix docstrings for pandas.Period.month indentation

* Fix docstrings for pandas.Period.month

* Fix docstrings for pandas.Period.asfreq SA01

* Fix docstrings for pandas.Period.asfreq

* Fix docstrings ci error cython-lint

* Fix docstrings pre-commit.cierror

* Fix docstring validation error

* Fix docstring validation errors

Siniade committed:
DOC: Enforce Numpy Docstring Validation for freqstr, nanos, and rule_code methods (#59475)

* DOC: Enforce Numpy Docstring Validation for freqstr, nanos, and rule_code methods of pandas.tseries.offsets Week, WeekOfMonth, YearBegin, and YearEnd classes.

* DOC: Enforce Numpy Docstring Validation for freqstr, nanos, and rule_code methods of pandas.tseries.offsets Week, WeekOfMonth, YearBegin, and YearEnd classes.

* Fix precommit errors - Add 2 spaces before inline comment

* Fix precommit errors - Remove unwanted language

* Update code_checks.sh to remove methods that don't fail anymore.

* Indented See Also strings to avoid using # noqa

* Correct for code check error.

* Removed duplicate from code_checks.sh.

* Correct code_checks.sh.

* Correct code_checks.sh.

* Update pandas/_libs/tslibs/offsets.pyx

Delete two examples from See Also for nanos

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/_libs/tslibs/offsets.pyx

Update nanos summary

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/_libs/tslibs/offsets.pyx

Delete two examples from See Also for nanos

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
REF (string): remove _str_na_value (#59515)

* REF (string): remove _str_na_value

* mypy fixup

jbrockmendel committed:
REF (string): move ArrowStringArrayNumpySemantics methods to base class (#59501)

* REF: move ArrowStringArrayNumpySemantics methods to parent class

* REF: move methods to ArrowStringArray

* mypy fixup

* Fix incorrect double-unpacking

* move methods to subclass

matsidzi committed:
TST: Added test for date_range for bug GH#57456 (#59519)

TST: Added test for date_range for bug #57456

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

uditbaliyan committed:
Fix docstrings for pandas.Period.year SA01 (#59525)

* Fix docstrings for pandas.Period.year SA01

* Fix docstrings for pandas.Period.year SA01 Validate Docstrings err

* Fix docstrings for pandas.Period.year SA01 Validate Docstrings err

* Removed dateparseerror example from period.year

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
API (string): return str dtype for .dt methods, DatetimeIndex methods (#59526)

* API (string): return str dtype for .dt methods, DatetimeIndex methods

* mypy fixup

MarcoGorelli committed:
DOCS: remove pandas-coverage app (#59536)

[skip-ci] docs: remove pandas-coverage app

phofl committed:
CI: Fix ci for numpy 2 failures (#59545)

WillAyd committed:
Fix get_datetimestruct_days overflow (#56001)

* remove pandas/tests/frame UB

* updated macro with filename / line num

* fix typo

* stay with PD_CHECK_OVERFLOW

* updates

* test fixes

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

EduardAkhmetshin committed:
TST: Test non-nanosecond datetimes in PyArrow Parquet dataframes (#59393)

* Add test_non_nanosecond_timestamps

* Fix lint errors

* Remove detailed comment; use temp path helper

* Use temp_file

* Apply suggestions from code review

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Remove extra empty line

* Skip test in minimal version env

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

matsidzi committed:
DOC: Fixed bug number in what's new v3.0.0 (#59539)

Fixed bug number in what's new v3.0.0

abhinav-thimma committed:
DOC: Fix Numpy Docstring validation errors in pandas.api.extensions.ExtensionArray (#59540)

* Fix SA01 for isna

* Fix See Also for nbytes

* fix See also for ndim

* Fix ravel

* fis return value of take

* remove type ignore

---------

Co-authored-by: Abhinav Thimma <athimma2@illinois.edu>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

uditbaliyan committed:
Fix docstring timestamps#59458 (#59533)

* Fix docstrings for pandas.Period.year SA01

* Fix docstrings for pandas.Period.year SA01 Validate Docstrings err

* Fix docstrings for pandas.Period.year SA01 Validate Docstrings err

* Fix docstrings for pandas.timestamps SA01 PR01

* This reverts commit f52c61080a331597197711465742ace1bdfcd1dd.

* Revert "Removed dateparseerror example from period.year"

This reverts commit cdf36b2faa6c95848d6210c801ee287f0b7084fd.

* fixing merge conflict

* fix pre-commit.ci err

* fix pre-commit.ci err

* fix Docstring validation err

* fix Docstring validation err

* fix Docstring validation err

* fix timestamp Docstring validation err

* pandas/plotting/_matplotlib/core.py:549: error: Unused "type: ignore" comment [unused-ignore]

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

aram-cedarwood committed:
BUG: `query` on columns with characters like # in its name (#59296)

mroeschke committed:
CI: Uninstall nomkl & 32 bit Interval tests (#59553)

* undo numpy 2 changes?

* some interval 32 bit tests working

* Revert "undo numpy 2 changes?"

This reverts commit 39ce2229a96406edac107fc897e807251d364e2b.

* nomkl?

* nomkl?

* Update .github/actions/build_pandas/action.yml

* grep for nomkl

* xfail WASM

* Reverse condition

jsngn committed:
DOC: Fix docstrings for Timestamp: strptime, timetz, to_datetime64, to_julian_date (#59569)

* strptime

* strptime + timestamp

* timetuple

* timetz

* to_datetime64

* to_julian_date

* timetz

* code_checks

* duplicate see also

messense committed:
PERF: avoid calling `DataFrame.dtypes` in loop (#59573)

fbourgey committed:
Series replace error (#59552)

* changed AttributeError to ValueError

* added test for replace method from dict_like to dict_like for a pd.Series

* added entry in whatsnew.rst

* Update doc/source/whatsnew/v3.0.0.rst

removed rst entry

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/core/generic.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/core/generic.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* updated msg test function

* breaking line for 88 instead of 89 characters

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jorisvandenbossche committed:
String dtype: still return nullable NA-variant in object inference (`maybe_converts_object`) if requested (#59487)

* String dtype: maybe_converts_object give precedence to nullable dtype

* update datetimelike input validation

* update tests and remove xfails

* explicitly test pd.array() behaviour (remove xfail)

* fixup allow_2d

* undo changes related to datetimelike input validation

* fix test for str on current main

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

maymunashah committed:
DOC: Clarify date_format usage in read_csv documentation specifically… (#59566)

DOC: Clarify date_format usage in read_csv documentation specifically pandas/io/parsers/readers.py issue #59557

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

rob-sil committed:
BUG: Handle nonexistent end dates when resampling (#59471)

jorisvandenbossche committed:
String dtype: fix pyarrow-based IO + update tests (#59478)

jorisvandenbossche committed:
TST: link to correct downstream fastparquet bug (#59581)

AntiKnot committed:
BUG: Fix Ability to set both color and style in pandas plotting (#59574)

* BUG: Fix plotting set color style dict sametime

* BUG: chore pre-commit inconsistent-namespace-usage

* BUG: Chore Conditional block

* Update doc/source/whatsnew/v3.0.0.rst

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* BUG: Chore pre-commit sort whatsnew entries alphabetically

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

KevsterAmp committed:
ENH: Add merge type validation on `pandas.merge` (#59435)

* add merge type validation on pandas.merge

* add ENH to latest whatsnew doc

* move merge type validation to _MergeOperation class

* add tests on merge validation; add asof as valid mergetype

* change merge_type from tuple to dict

* change ValueError message to updated change

* change Error message on merge type

* update test error messages for merge type errors

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

awojno-bloomberg committed:
DOC: Update Series docs for Series.update (#59579)

DOC: Update Series docs

pandas.Series.update

jbrockmendel committed:
REF (string): avoid copy in StringArray factorize (#59551)

* REF: avoid copy in StringArray factorize

* mypy fixup

* un-xfail

serl committed:
Fix typo in contributing_environment.rst (#59584)

WillAyd committed:
Use shorthand notation for pip config-settings (#59583)

Pranav-Wadhwa committed:
Clarify documentation for asfreq (#59589)

hlakams committed:
DOCS: fix docstring validation errors for pandas.Series (#59602)

* DOCS: pandas.Series.prod + pandas.Series.product RT03 docstring validation error fixed

* DOCS: pandas.Series.pop SA01 + pandas.Series.reorder_levels RT03/SA01 docstring validation error fixed

* DOCS: pandas.Series.list.__getitem__ + pandas.Series.list.flatten + pandas.Series.list.len SA01 docstring validation error fixed

* DOCS: pandas.Series.sparse.density SA01 docstring validation error fixed

* DOCS: pandas.Series.gt + pandas.Series.lt + pandas.Series.ne SA01 docstring validation error fixed

* linting issues leftover from docstring validation fixes resolved

fbourgey committed:
Add example same correlation in pandas.Series.corr documentation (#59591)

ankit-dhokariya committed:
DOC: Enforce Numpy Docstring Validation (Issue #59458) (#59590)

* adding docstring for pandas.Timestamp.day property

* fixing type annotation

ivonastojanovic committed:
DOCS: fix docstring validation errors for pandas.Series (#59596)

Fixes:
-i "pandas.Series.str.match RT03" \
-i "pandas.Series.str.normalize RT03,SA01" \
-i "pandas.Series.str.repeat SA01" \
-i "pandas.Series.str.replace SA01" \

fangchenli committed:
MAINT: update vendored version util from packaging (#59558)

* MAINT: update vendored version util from packaging

* fix docstring

* fix docstring

* skip docstring validation

* ignore

* fix validation ignore

* remove docstring

* rollback

* add comments

ivonastojanovic committed:
DOC: Enforce Numpy Docstring Validation | pandas.api.extensions.ExtensionArray (#59407)

* Fix pandas.api.extensions.ExtensionArray._pad_or_backfill

Add 'limit_area' parameter, return value description and 'See Also' section

* Fix pandas.api.extensions.ExtensionArray._reduce

Add return value description and 'See Also' section

* Fix pandas.api.extensions.ExtensionArray._values_for_factorize

Add 'See Also' section

* Fix pandas.api.extensions.ExtensionArray.astype

Add 'See Also' section

* Fix pandas.api.extensions.ExtensionArray.dropna

Add return value description and 'See Also' section

* Fix pandas.api.extensions.ExtensionArray.dtype

Add 'See Also' section

alex-lliu committed:
DOCS: fix docstring validation errors for pandas.Series (#59600)

* DOCS: fix docstring validation errors for pandas.Series

* DOCS: fix underline length

wenchen-cai committed:
DOCS: fix docstring validation errors for pandas.Series.str (#59597)

abhinav-thimma committed:
DOC: Fix Numpy Docstring errors in pandas.api.extensions.ExtensionArray (#59605)

* fix duplicated

* fix fillna

* fix insert

* fix isin

* fix tolist

* fix unique

* fix view

---------

Co-authored-by: Abhinav Thimma <athimma2@illinois.edu>

ktseng4096 committed:
DOC: add See Also section to groupby.DataFrameGroupBy.prod (#59599)

* Update Groupby.prod

* update code_check list

* remove extra spaces

* fix errors

* ruff formatting

MarcoGorelli committed:
ENH: support Arrow PyCapsule Interface on Series for export (#59587)

* ENH: support Arrow PyCapsule Interface on Series for export

* simplify

* simplify

mroeschke committed:
REF: Minimize operations in recode_for_groupby (#59618)

matiaslindgren committed:
BUG: allow None as name in multi-index during join (#59546)

* allow None as name in multi-index

* update whatsnew

* add unit test for none label joins

* move bugfix note under Reshaping

* Update doc/source/whatsnew/v3.0.0.rst

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

KevsterAmp committed:
PERF: Performance Improvement on `DataFrame.to_csv()` when `index=False` (#59608)

* add alternative ix when self.nlevel is 0

* add to latest whatsnew

* change np.full to np.empty

hlakams committed:
DOC: fix docstring validation errors for pandas.io.formats.style.Styler (#59607)

* DOC: all pandas.io.formats.style.Styler docstring validation errors fixed

* DOCS: base to_excel docstring template extended for pandas.io.formats.style.Styler.to_excel

maushumee committed:
BUG: Fix for pandas.to_datetime reports incorrect index when failing (#59594)

* Remove index from error

* Fix test failures

* Add entry to docs/source/whatsnew

* Update pandas/tests/tools/test_to_datetime.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/tests/tools/test_to_datetime.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jorisvandenbossche committed:
String dtype: avoid surfacing pyarrow exception in binary operations (#59610)

ankit-dhokariya committed:
DOC: Enforce Numpy Docstring Validation (Issue #59458) (#59622)

* adding docstring for Timestamp properties

* updating code_checks.sh

* removing extra line

* fixing code_checks.sh

james-magee committed:
DOC: fix docstring validation errors for pandas.Series (#59621)

* fixed Series.std and Series.sem docstrings via make_doc

* removed Series.sem, Series.std from ci/code_checks.sh

* fixed errors from pre-commit

* ran pre-commit

astronights committed:
ENH: Enhancement for spss kwargs (#56434)

* Bug fix for spss kwargs

* Update to whatsnew with bug fix details

* Black formatting fix

* Fix for whatsnew rst pre-commit

* Sort whatsnew alphabetically

* Fixing alphabetical sorting to whatsnew

* Sorting whatsnew

* Fixing kwargs

* Fixing tests

* Test fixes

* Minor change in kwargs

* Resolving PR comments

* Resolving mypy issue

* Docstring update for kwargs mypy

* Updates based on PR comments

* Fixing kwargs types

* Finalizing updates based on suggestions

* Updating to try to fix other random test fails

* Unrelated test fix

* Test fixes for dims/sizes

* Updates to whatsnew

* Case fix

* Update to reword to enhancements

jorisvandenbossche committed:
DOC: Add whatsnew for 2.3.0 (#59625)

* DOC: Add whatsnew for 2.3.0

* fix duplicate label

jbrockmendel committed:
BUG (string): str.replace with negative n (#59628)

* BUG (string): str.replace with negative n

* update GH ref

jbrockmendel committed:
PERF: avoid unnecessary copy (#59630)

pol-rius committed:
DOC: fix docstring validation errors for pandas.Series (#59633)

* fixed Series.sparse and Series.sparse.to_coo and removed from ci/code_checks.sh

* fixed errors from pre-commit

* fixed errors from pre-commit

hlakams committed:
DOC: fix docstring validation errors for pandas.PeriodIndex (#59638)

* DOC: pandas.PeriodIndex GL08, PR07, RT03, and SA01 docstring validation errors fixed

* DOC: pandas.Series.dt.qyear docstring validation error addressed by existing series.PeriodIndex.qyear docstring validation fix

hlakams committed:
DOC: fix docstring validation errors for pandas.core.window (#59634)

* DOCS: pandas.core.window SA01 docstring validation error fixed

* DOC: pandas.core.window SA01 docstring validation error linting fixes

* DOC: pandas.core.window.rolling.Window.var PR01 docstring validation fix

jbrockmendel committed:
TST (string): fix xfailed groupby value_counts tests (#59632)

jbrockmendel committed:
REF (string): rename result converter methods (#59626)

mroeschke committed:
REF: Consistently use `_reset_cache` for cache clearing (#59617)

Standardize cache clearing

Tolker-KU committed:
PERF: Improve efficiency of `BlockValuesRefs` (#59598)

* Improve efficency of _libs.internal

* Re-add python version of ._rebuild_blknos_and_blklocs

specialkapa committed:
Doc/59592/fix docstring validation errors (#59611)

* adding extended summary, populating return description and adding see also section

* populating return description section

* populating return description section

* fixing zfil docstrign as per PR #59597

* removing methods whose docstrign has been fixed from code_checks.sh script

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

Tolker-KU committed:
PERF: Improve efficiency of `BlockValuesRefs._clear_dead_references(...)` (#59643)

Improve performance of BlockValuesRefs._clear_dead_references(...)

jbrockmendel committed:
TST (string) fix xfailed groupby tests (3) (#59642)

* TST (string) fix xfailed groupby tests (3)

* TST: non-pyarrow build

uditbaliyan committed:
Docstring timestamps#59458 (#59644)

* fix pandas.Timestamp.fold GL08

* fix pandas.Timestamp.fold GL08

* remove i- pandas.Timestamp.fold GL08

* fix pre-commit error

* fix docstring error

jbrockmendel committed:
REF (string): de-duplicate str_endswith, startswith (#59568)

callumfrederiksen committed:
Fixed Typos #59640 (#59651)

* Fixed Typos

* Changes requested

tuhinsharma121 committed:
DOC: fix RT03,SA01,ES01 for pandas.MultiIndex.reorder_levels (#59658)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.Period.freqstr (#59661)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.NaT (#59660)

musvaage committed:
fix typos (#59665)

typos

mattwang44 committed:
DOC: Remove unnecessary parentheses in `:func:` & `:meth:` role (#59663)

remove unnecessary parentheses in func & meth role from doc

mroeschke committed:
REF: Use range over arange for iterators (#59619)

musvaage committed:
fix typo (#59669)

typo

jbrockmendel committed:
DEPR (string): non-bool na for obj.str.contains (#59615)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

jsngn committed:
TST: Add SparseArray _arith_method test to fail on incorrect comparison operator (#59648)

* Test

* Match

* Rerun GitHub actions

* Initialize inside func

tuhinsharma121 committed:
DOC: fix PR01,SA01 for pandas.Period.strftime (#59673)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.PeriodDtype (#59674)

tuhinsharma121 committed:
DOC: fix PR07 for pandas.RangeIndex (#59675)

tuhinsharma121 committed:
DOC: fix PR01,SA01,ES01 for pandas.errors.AbstractMethodError (#59679)

mroeschke committed:
REF: Use numpy methods instead of np.array (#59672)

jorisvandenbossche committed:
TST (string dtype): fix and clean up arrow roundtrip tests (#59678)

* TST (string dtype): fix and clean up arrow roundtrip tests

* fix using_infer_string

jbrockmendel committed:
API (string): str.center with pyarrow-backed string dtype (#59624)

mroeschke committed:
PERF: Only clear cached .levels when setting MultiIndex.names (#59578)

* PERF: Only clear cached .levels when setting MultiIndex.names

* whatsnew number

ivonastojanovic committed:
DOC: Add 'See Also'and return value description for pandas.api.interchang.from_dataframe (#59403)

DOCS: Add 'See Also'and return value description for pandas.api.interchange.from_dataframe

michaelmannino committed:
ENH: DataFrame.plot.scatter argument `c` now accepts a column of strings, where rows with the same string are colored identically (#59239)

serl committed:
Fix obj arguments in assertions (#59460)

* Fix obj arguments in assertions

* Add test for assert_interval_array_equal

* Still say MultiIndex in some cases

* Be a little braver and change the original obj definition

mimistiv committed:
Fix docstring timestamps (Issue #59458) (#59701)

add value docstring

jbrockmendel committed:
REF (string): de-duplicate str_isfoo methods (#59705)

jbrockmendel committed:
TST (string): copy/view tests (#59702)

doshi-kevin committed:
Changed to GroupBy (#59699)

* Changed to GroupBy

* Update pandas/core/groupby/groupby.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/core/groupby/groupby.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
TST (string): more targeted xfails in test_string.py (#59703)

* TST (string): more targeted xfails in test_string.py

* Fix no-pyarrow test

* Update pandas/tests/extension/test_string.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/tests/extension/test_string.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

uditbaliyan committed:
Fix part of docstring timedelta index#59698 (#59707)

* fix pandas.Timestamp.fold GL08

* fix pandas.Timestamp.fold GL08

* remove i- pandas.Timestamp.fold GL08

* fix pre-commit error

* fix docstring error

* add see also for ceil,floor,asm8,round

* remove ceil,floor,asm8,round from code_checks.sh

mroeschke committed:
PERF: CategoricalDtype.update_dtype (#59647)

* PERF: CategoricalDtype.update_dtype

* Add whatsnew number add comment

* Fix unit test

* short circut only for the dtype

jsngn committed:
TST: Update BooleanArray _logical_method test to fail on incorrect length comparison operator (#59708)

Test

jbrockmendel committed:
REF (string): de-duplicate _str_contains (#59709)

* REF: de-duplicate _str_contains

* pyright ignore

amanlai committed:
DOC: Fix some docstring validation errors #59698 (#59713)

* fix some docstring errors

* removed trailing whitespace

* pd.Series.dt.microseconds has the same documentation as pd.TimedeltaIndex.microseconds and SA01 was cleared for both in the previous commit

deekapila committed:
DOC: Clarify docs for df.to_sql (#59727)

jbrockmendel committed:
BUG (string): ArrowStringArray.find corner cases (#59562)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.Period.to_timestamp (#59730)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.Timedelta.view (#59733)

jorisvandenbossche committed:
String dtype: implement _get_common_dtype (#59682)

* String dtype: implement _get_common_dtype

* add specific tests

* try fix typing

* try fix typing

* suppress typing error

* support numpy 2.0 string

* fix typo

natmokval committed:
DOC: move `idxmin` and `idxmax` docs from core/shared_docs.py to core/frame.py (#59735)

move idxmin, idxmax docstring from shared_docs.py to frame.py

ammar-qazi committed:
Resolves #59670 by documenting that DataFrame.from_records()'s columns filters (includes) data.  (#59723)

Update frames.py to factor in explain columns reordering

fbourgey committed:
Missing source link (#59549)

* merged DataFrame.index and DataFrame.columns with other Axes section.

* small clean for DataFrame.columns

* reverted frame.rst file

KevsterAmp commented:
CIs are failing, not sure if this is the right target branch to open a PR

rhshadrach commented:
@lithomas1 - am I right in guessing that there is likely not to be any more releases from the `2.2.x` branch? If that's the case, this PR should target `2.3.x`.

matiaslindgren committed:
BUG: Fix inconsistent pivot table subaggregation when index is None (#59629)

phofl committed:
TST/BUG (string dtype): Fix and adjust indexes string tests (#59544)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

phofl committed:
TST (string dtype): Adjust indexing string tests (#59541)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

dependabot[bot] committed:
Bump deadsnakes/action from 3.1.0 to 3.2.0 (#59757)

Bumps [deadsnakes/action](https://github.com/deadsnakes/action) from 3.1.0 to 3.2.0.
- [Release notes](https://github.com/deadsnakes/action/releases)
- [Commits](https://github.com/deadsnakes/action/compare/v3.1.0...v3.2.0)

---
updated-dependencies:
- dependency-name: deadsnakes/action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jorisvandenbossche committed:
TST (string dtype): adjust pandas/tests/reshape tests (#59762)

abhinav-thimma committed:
Fix docs for api.types (#59753)

* Fix is_bool

* Fix is_categorical_dtype

* Fix is_complex

* Fix is_complex_dtype

* Fix is_datetime64_dtype

* Fix is_datetime64_ns_dtype

* Fix is_datetime64tz_dtype

---------

Co-authored-by: Abhinav Thimma <athimma2@illinois.edu>

jsngn committed:
TST: Update IntervalArray min/max test to fail on changed default skipna (#59747)

Test

ivanpan0626 committed:
DOCS: fix docstring validation errors for groupby.DataFrameGroupBy.filter, groupby.SeriesGroupBy.filter (#59742)

* DOCS: fix docstring validation errors for pandas.core.groupby.DataFrameGroupBy.filter

DOC string fix for both groupby.DataFrameGroupBy.filter and groupby.SeriesGroupBy.filter

* Update generic.py

* Update generic.py

* Update generic.py

* quickfix

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.RangeIndex.stop (#59729)

* DOC: fix SA01,ES01 for pandas.RangeIndex.stop

* remove superfluous description of RangeIndex

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.RangeIndex.start (#59728)

* DOC: fix SA01,ES01 for pandas.RangeIndex.start

* remove superfluous description of RangeIndex

MarcoGorelli committed:
fix: use fastpath for PyCapsule export when starting from pyarrow-backed Series, respect requested_schema (#59683)

* fix: use fastpath for PyCapsule export when starting from pyarrow-backed Series, respect requested_schema

* simplify

* stringdtype test

KatsiarynaDzibrova committed:
DOC: Fix pandas.Series.dt seconds, nanoseconds GL08, SA01 (#59582)

* fix pandas.Series.dt.freq

* fix seconds, nanoseconds, microseconds

* remove fixed objects from code_checks.sh

* Remove Timedelta Index checks

* fix freq example

* remove freq

* bring back microseconds

jorisvandenbossche committed:
BUG (string dtype): fix inplace mutation with copy=False in ensure_string_array (#59756)

* BUG (string dtype): fix inplace mutation with copy=False in ensure_string_array

* update

jorisvandenbossche committed:
TST (string dtype): remove usage of 'string[pyarrow_numpy]' alias (#59758)

aaronchucarroll committed:
ENH: Add dtype argument to StringMethods get_dummies() (#59577)

jbrockmendel committed:
BUG (string): Series.str.slice with negative step (#59724)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

jorisvandenbossche committed:
String dtype: remove fallback Perfomance warnings for string methods (#59760)

ammar-qazi committed:
DOC: Add docstring for Extensionarray interpolate (#59749)

* Update docstring of Extensionarray.interpolate

* Remove Extensionarray.interpolate from code_checks.sh

* Resolving pre-commit errors

* Resolving pre-commit errors 2

* Resolved ruff formatting error

* Fix issues after review

jbrockmendel committed:
REF (string): de-duplicate ArrowStringArray methods (#59555)

sshu2017 committed:
Fix/na_values_GH59303 (#59755)

* fixed GH#59303

* pre-commit done

* updated v3.0.0.rst

* sort my entry in v3.0.0.rst

* changes based on comments on PR

* reformat long lines

* reformat test_na_values.py

* reformat test_na_values.py again

jorisvandenbossche committed:
BUG: avoid triggering numpy deprecation warning in assert functions for nested array with empty array/list (#59778)

ktseng4096 committed:
DOC: Update GroupBy docstrings with See Also requirements (#59748)

* update groupby docstrings

* fix function name

jorisvandenbossche committed:
BUG/API (string dtype): return float dtype for series[str].rank() (#59768)

* BUG/API (string dtype): return float dtype for series[str].rank()

* update frame tests

* add whatsnew

* correct whatsnew note

jorisvandenbossche committed:
String dtype: fix isin() values handling for python storage (#59759)

* String dtype: fix isin() values handling for python storage

* address feedback

tuhinsharma121 committed:
DOC: fix SA01 for pandas.NA (#59787)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.Timedelta.components (#59799)

ammar-qazi committed:
Update ExtensionArray.interpolate to remove outdated method of pad (#59798)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.PeriodDtype.freq (#59796)

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.Timedelta.total_seconds (#59800)

* DOC: fix SA01,ES01 for pandas.Timedelta.total_seconds

* DOC: fix SA01,ES01 for pandas.Timedelta.total_seconds

tuhinsharma121 committed:
DOC: fix SA01 for pandas.api.types.is_array_like (#59802)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.api.types.is_integer_dtype (#59803)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.arrays.FloatingArray (#59804)

twoertwein committed:
WEB: update list of (in)active core devs (#59808)

jorisvandenbossche committed:
BUG (CoW): fix reference tracking in replace_list with None (#59807)

jorisvandenbossche committed:
DOC: add whatsnew for v2.2.3 (#59811)

* DOC: add whatsnew for v2.2.3

* fix warning

---------

Co-authored-by: Thomas Li <47963215+lithomas1@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01 for pandas.errors.MergeError (#59805)

* DOC: fix SA01 for pandas.errors.MergeError

* DOC: fix SA01 for pandas.errors.MergeError

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

jorisvandenbossche committed:
String dtype: allow string dtype in query/eval with default numexpr engine (#59810)

String dtype: allow string dtype in query/eval with default mumexpr engine

lysnikolaou committed:
BUG: Remove np._get_promotion_state usage (#59818)

lithomas1 committed:
CI: Debug failing ARM builds (#59813)

* try bumping cython?

* maybe pinning numpy helps?

* skip tests

* Update test_sparse.py

* go for green

* Update test_sparse.py

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.tseries.offsets.WeekOfMonth (#59834)

ms041223 committed:
DOC: Adding ArcticDB to the ecosystem.md page (#59830)

* adding ArcticDB to the ecosystem.md page

* Update web/pandas/community/ecosystem.md

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* making pandas lower case

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

lithomas1 committed:
BLD: Fix bad Cython annotation (#59836)

jorisvandenbossche committed:
BLD/RLS: build wheels with released numpy/cython for Python 3.13 (#59819)

lithomas1 committed:
BLD: Final release prep for 2.2.3 (#59840)

* BLD: Final release prep

* change back perms

* debug

* try to fix license addition

* silence stable version warning?

lithomas1 committed:
BLD: Build wheels for Python 3.13 on aarch64 as well (#59847)

* BLD: Build wheels for Python 3.13 on aarch64 as well

* some fixups

* another typo

tuhinsharma121 committed:
DOC: fix RT03 for pandas.core.groupby.DataFrameGroupBy.hist (#59870)

fawazahmed0 committed:
BUG: Fix precision loss in read_json (#59284)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.api.types.is_int64_dtype (#59862)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.api.types.is_float_dtype (#59861)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.Series.sparse.sp_values (#59859)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.Series.sparse.fill_value (#59858)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.tseries.offsets.SemiMonthEnd (#59856)

DOC: fix SA01 for pandas.tseries.offsets.SemiMonthEnd

jorisvandenbossche committed:
String dtype: map builtin str alias to StringDtype (#59685)

* String dtype: map builtin str alias to StringDtype

* fix tests

* fix datetimelike astype and more tests

* remove xfails

* try fix typing

* fix copy_view tests

* fix remaining tests with infer_string enabled

* ignore typing issue for now

* move to common.py

* simplify Categorical._str_get_dummies

* small cleanup

* fix ensure_string_array to not modify extension arrays inplace

* fix ensure_string_array once more + fix is_extension_array_dtype for str

* still xfail TestArrowArray::test_astype_str when not using infer_string

* ensure maybe_convert_objects copies object dtype input array when inferring StringDtype

* update test_1d_object_array_does_not_copy test

* update constructor copy test + do not copy in maybe_convert_objects?

* skip str.get_dummies test for now

* use pandas_dtype() instead of registry.find

* fix corner cases for calling pandas_dtype

* add TODO comment in ensure_string_array

tuhinsharma121 committed:
DOC: fix SA01 for pandas.api.types.is_interval_dtype (#59863)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.api.types.is_list_like (#59864)

musvaage committed:
typo (#59852)

Commit:
BUG: Fix from_records() column reorder issue, if columns!=None use passed param (#59717) (#59809)

* BUG: Fix columns param reorder issue - if columns!=None, use passed param (#59717)

* Add tests for to_arrays()

* Fix import order with isort

* fix sort

* Update datatype to int32

* Fis test

* Revert commit

* Add test for DaaFrame.from_records()

* Apply comments

* Delete test_to_arrays.py

tuhinsharma121 committed:
DOC: fix SA01 for pandas.arrays.BooleanArray (#59866)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.api.types.is_object_dtype (#59865)

tuhinsharma121 committed:
DOC: fix RT03 for pandas.date_range (#59868)

tuhinsharma121 committed:
DOC: fix RT03, ES01 for pandas.core.resample.Resampler.ffill (#59871)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.arrays.IntervalArray.mid (#59867)

* DOC: fix SA01, ES01 for pandas.arrays.IntervalArray.mid

* DOC: add double backticks for sphinx compatibility

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.RangeIndex.step (#59857)

* DOC: fix SA01, ES01 for pandas.RangeIndex.step

* DOC: add double backticks for sphinx compatibility

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.Timedelta.to_timedelta64 (#59860)

* DOC: fix SA01, ES01 for pandas.Timedelta.to_timedelta64

* DOC: add double backticks for sphinx compatibility

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.errors.EmptyDataError (#59872)

JonathanMarriott committed:
DOC: Fix inconsistent and incomplete documentation of `pandas.eval` (#59855)

* Improve content and organisation of eval documentation

* Link to pd.eval in pd.DataFrame.query

* Correct name for `//` is floor division

* Include arctan2

Co-authored-by: Xiao Yuan <yuanx749@gmail.com>

---------

Co-authored-by: Xiao Yuan <yuanx749@gmail.com>

jorisvandenbossche committed:
String dtype: allow string dtype for non-raw apply with numba engine (#59854)

* String dtype: allow string dtype for non-raw apply with numba engine

* remove xfails

* clean-up

nemausus committed:
ENH: Add kwargs to Series.map (#59843)

Co-authored-by: Naresh Kumar <naresh.kumar@snowflake.com>

jbrockmendel committed:
REF: pass dtype explicitly to _from_sequence inside pd.array (#59773)

REF: pass dtype explicitly to _from_sequence

jbrockmendel committed:
REF: pass dtype explicitly to _from_sequence (#59774)

mroeschke committed:
DOC: Emphasize team managed pandas in installation docs (#59822)

* DOC: Emphasize team managed pandas in installation docs

* grammar

KevsterAmp commented:
Gotcha, retargeting to branch 2.3.x

mroeschke committed:
CI/TST: Check for tzset in set_timezone (#59893)

* CI/TST: Check for tzset in set_timezone

* adjust test message

mroeschke committed:
DOC: Recommend conda from miniforge for contributing environment (#59894)

rhshadrach committed:
CI: Pin micromamba to 1.x (#59912)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.Series.sparse.npoints (#59896)

* DOC: fix SA01, ES01 for pandas.Series.sparse.npoints

* Update pandas/core/arrays/sparse/array.py

tuhinsharma121 committed:
DOC: fix RT03, ES01 for pandas.core.groupby.DataFrameGroupBy.agg and pandas.core.groupby.DataFrameGroupBy.aggregate (#59869)

* DOC: add double backticks for sphinx compatibility

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>

* DOC: remove _agg_template_frame

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>
Co-authored-by: rhshadrach <rhshadrach@users.noreply.github.com>

* DOC: fix RT03, ES01 for pandas.core.groupby.DataFrameGroupBy.aggregate

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>
Co-authored-by: rhshadrach <rhshadrach@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix RT03, ES01 for pandas.core.groupby.SeriesGroupBy.agg and pandas.core.groupby.SeriesGroupBy.aggregate (#59898)

* DOC: fix RT03, ES01 for pandas.core.groupby.SeriesGroupBy.agg

* DOC: remove _agg_template_series

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>
Co-authored-by: rhshadrach <rhshadrach@users.noreply.github.com>

* DOC: remove _agg_template_series

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>
Co-authored-by: rhshadrach <rhshadrach@users.noreply.github.com>

* DOC: remove _agg_template_seris

---------

Co-authored-by: mroeschke <mroeschke@users.noreply.github.com>
Co-authored-by: rhshadrach <rhshadrach@users.noreply.github.com>

saldanhad committed:
DOC: fix docstrings for multiple api.types methods (#59920)

fix docstrings for api.types

gameofby committed:
DOC: the table name should be `air_quality_parameters` rather than `air_quality_parameters_name` (#59918)

saldanhad committed:
DOC: Separate out examples - pandas.str.is methods (#59850)

saldanhad committed:
DOC: fix pandas.TimedeltaIndex.to_pytimedelta RT03,SA01 (#59914)

* update docstrings

* Update pandas/core/arrays/timedeltas.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.errors.ClosedFileError (#59924)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.errors.OutOfBoundsDatetime (#59925)

fbourgey committed:
Fix docstring Timedelta.to_timedelta64 SA01, Timedelta.total_seconds SA01, Timedelta.view SA01 (#59719)

* Add 'See Also' section for Timedelta.to_timedelta64

* Fix SA01 for Timedelta.total_seconds()

* Fix SA01 for Timedelta.view

* Add space

* Fix test_nat_doc_strings

* Revert "Fix test_nat_doc_strings"

This reverts commit 9d0965805daa2dbd02eaa1878858cfb0eb97df02.

* Match doc of total_seconds method in nattype.pyx

qaiser42 committed:
Fix typo in 10min.rst (#59921)

cdce8p committed:
BLD: Fix armv7 build (#59906)

fbourgey committed:
DOC: Fix intro to datastructures Series constructor behavior (#59793)

Petroncini committed:
BUG: groupby().any() returns true for groups with timedelta all NaT (#59782)

jorisvandenbossche committed:
CI: Run jobs on 2.3.x branch (#59939)

cdce8p committed:
Include Python.h first (#59929)

FuzzyParrabellum committed:
DOC: Fix docstring of pandas.Series.compare list indent formatting (#59911)

Co-authored-by: rdzantoine.pro@gmail.com <rdzantoine.pro@gmail.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

KevsterAmp commented:
@rhshadrach - moved it to 2.3.x. Do I need to add any other changes here? Thanks

yuanx749 committed:
BUG: pd.eval with engine="numexpr" fails with float division (#59907)

* BUG: pd.eval with engine="numexpr" fails with float division

* Add skip

* Add whatsnew

* update

mroeschke committed:
CLN: indexes/base.py (#59928)

CLN: indexes.base.py

rhshadrach commented:
@KevsterAmp - can you merge main to kick off the CI again

saldanhad committed:
BUG: fix html float display (#59930)

* fix html display float/strings

* add test under io, update whatsnew

* fix linting

* changes to fix floats only

* Revert "fix linting"

This reverts commit 1061442e0a1cf8f745b0863762f2aa023d388336.

* test script for float format

* remove nbsp implementation, keep floats

* Trigger CI

* implement changes post review

* lint check

* update test_formats.py

* rfc test_format.py

* update test cases

saldanhad committed:
BUG: fix nbsp for html formatting (#59964)

* nbsp for strings

* update changes post review

saldanhad committed:
BUG: fix treatment of NaNs when .apply() function is used on categorical columns. (#59966)

* remove action=ignore for .apply() on cat dtype

* add PR reference in comments

* fix pytest linting

* refac failing test_series_apply.py

* Trigger CI

* changes post review

* rephrase change log

janezd committed:
BUG: Fix SparseFrameAccessor.to_dense return type (#59967)

* BUG: Fix SparseFrameAccessor.to_dense return type

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

invalidarg committed:
BUG: CSS strings truncated at ":" (#59720)

* second item in tuple is no longer truncated at first colon

https://github.com/pandas-dev/pandas/issues/59623

* added testcase for maybe_convert_css_to_tuples

#59623

* maybe_convert_css_to_tuples() raises on strings without ":"

* fixed implicit str concatination

* Fixed raise on empty string

* Update test_style.py

* attr:; -> ("attr","")

Same behavior as before patch

* add test for "attr:;", ie empty value

* str concatenation in the test broke mypy

* revert explicit str concat

* Invalidarg patch black (#1)

* black test_style

* Update style_render.py

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix RT03,SA01 for pandas.period_range (#59958)

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.io.stata.StataReader.data_label (#59962)

yuanx749 committed:
BUG: fix to_numeric raises TypeError for Timedelta and Timestamp scalar (#59974)

* BUG: fix to_numeric raises TypeError for Timedelta and Timestamp scalar

* Add whatsnew

tuhinsharma121 committed:
DOC: fix SA01, ES01 for pandas.testing.assert_extension_array_equal (#59975)

rhshadrach committed:
API: value_counts to consistently maintain order of input (#59745)

* API: value_counts to consistently maintain order of input

* Docs

* Cleanup

* Test & docs fixups

* Refine whatsnew

* Refine whatsnew

tuhinsharma121 committed:
DOC: fix PR07,SA01,ES01 for pandas.Series.sparse.from_coo (#59980)

tuhinsharma121 committed:
DOC: fix PR01,SA01,ES01 for pandas.api.types.is_float (#59981)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.arrays.DatetimeArray (#59982)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.errors.SpecificationError (#59983)

tuhinsharma121 committed:
DOC: fix SA01 for pandas.errors.InvalidVersion (#59984)

StefRe committed:
DOC: Fix typos in plotting.table (#59986)

Fix typos in pandas.plotting.table docstring.

dependabot[bot] committed:
Bump mamba-org/setup-micromamba from 1 to 2 (#59988)

Bumps [mamba-org/setup-micromamba](https://github.com/mamba-org/setup-micromamba) from 1 to 2.
- [Release notes](https://github.com/mamba-org/setup-micromamba/releases)
- [Commits](https://github.com/mamba-org/setup-micromamba/compare/v1...v2)

---
updated-dependencies:
- dependency-name: mamba-org/setup-micromamba
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tuhinsharma121 committed:
DOC: fix RT03,SA01,ES01 for pandas.io.stata.StataReader.value_labels (#59991)

tuhinsharma121 committed:
DOC: fix RT03,SA01,ES01 for pandas.plotting.lag_plot (#59990)

randolf-scholz committed:
Typing: Added missing methods to `NaTType` stub (#59995)

added missing methods to NaTType stub

rhshadrach committed:
REGR: groupby.value_counts with all NA values (#59999)

* REGR: groupby.value_counts with all NA values

* Better implementation

saldanhad committed:
Doc: Update docstring for `dummy_na` parameter (#60000)

* update docstring for dummy_na parameter

* Update pandas/core/reshape/encoding.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

randolf-scholz committed:
Typing: More precise NaT stub (#60002)

* more precise NaT stub

* ruff format

* updated == and != to return literal

saldanhad committed:
DOC: Refactor _create_delegator_method using functools (#59878)

* add tag dt.to_timestamp, series.rst

* add doc strings for dt.to_timestamp

* update datetimes.py

* refactor _create_delegator_method to use functools wrap

* changes to accessor.py

* remove from code_checks.sh

* update code_checks.sh

* update code_checks.sh

* rewrite functools, adjust unit tests

* update change log

* remove dup entry

* update code_checks.sh

* update

* revert all dt related changes

* update series.rst

* update imports

* format use of functools import

albertvillanova committed:
Fix Styler docstring (#60001)

* Fix Styler docstring

* Remove blankspaces

rhshadrach commented:
@KevsterAmp - looks like the CI failures are related.

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#59998)

* [pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.5.0 → v0.6.9](https://github.com/astral-sh/ruff-pre-commit/compare/v0.5.0...v0.6.9)
- [github.com/jendrikseipp/vulture: v2.11 → v2.13](https://github.com/jendrikseipp/vulture/compare/v2.11...v2.13)
- [github.com/pre-commit/pre-commit-hooks: v4.6.0 → v5.0.0](https://github.com/pre-commit/pre-commit-hooks/compare/v4.6.0...v5.0.0)
- [github.com/asottile/pyupgrade: v3.16.0 → v3.17.0](https://github.com/asottile/pyupgrade/compare/v3.16.0...v3.17.0)
- [github.com/sphinx-contrib/sphinx-lint: v0.9.1 → v1.0.0](https://github.com/sphinx-contrib/sphinx-lint/compare/v0.9.1...v1.0.0)
- [github.com/pre-commit/mirrors-clang-format: v18.1.8 → v19.1.1](https://github.com/pre-commit/mirrors-clang-format/compare/v18.1.8...v19.1.1)

* Update .pre-commit-config.yaml

* fix style.ipynb, ignore some pylint

* pyupgrade

* Revert "pyupgrade"

This reverts commit b539c71009ff15769c501cf170ed9894a49ddcfb.

* don't bump pyupgrade

* Typo in random call

* Delete hidden cell

* Undo max/min rule from ruff

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

sunlight798 committed:
DOC: fix RT03,SA01,ES01 for pandas.io.stata.StataReader.variable_labels (#60008)

nahcol10 committed:
Programming Language :: Python :: 3.13 added to pyproject.toml (#59985)

* Programming Language :: Python :: 3.13 added to pyproject.toml

* pyproject.toml updated successfully

tuhinsharma121 committed:
DOC: fix SA01,ES01 for pandas.errors.CategoricalConversionWarning (#60011)

jorisvandenbossche committed:
String dtype: propagate NaNs as False in predicate methods (eg .str.startswith) (#59616)

jorisvandenbossche committed:
TST (string dtype): resolve all infer_string TODO/xfails in pandas/tests/arrays (#59686)

dependabot[bot] committed:
Bump pypa/cibuildwheel from 2.20.0 to 2.21.0 (#59816)

Bumps [pypa/cibuildwheel](https://github.com/pypa/cibuildwheel) from 2.20.0 to 2.21.0.
- [Release notes](https://github.com/pypa/cibuildwheel/releases)
- [Changelog](https://github.com/pypa/cibuildwheel/blob/main/docs/changelog.md)
- [Commits](https://github.com/pypa/cibuildwheel/compare/v2.20.0...v2.21.0)

---
updated-dependencies:
- dependency-name: pypa/cibuildwheel
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Thomas Li <47963215+lithomas1@users.noreply.github.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

mroeschke committed:
STY: Bump pyright, pyupgrade and mypy for new PEP-696 syntax (#60006)

* STY: Bump pyright, pyupgrade and mypy for new PEP-696 syntax

* Apply update

* fix & ignore failures

* another ignore

sparshsah committed:
DOC: Fix `GroupBy.ffill()` doc reference to rows (#60019)

Fix `groupby.ffill()` doc reference to rows

Wong2333 committed:
DOC: Fix title capitalization in documentation files (#32550) (#60021)

* DOC: Fix title capitalization in documentation file

* DOC: Fix title capitalization in documentation files

tuhinsharma121 committed:
DOC: fix SA01 for pandas.core.resample.Resampler.ohlc (#60036)

rhshadrach committed:
CI/TST: Update pyreadstat tests and pin xarray on CI (#60109)

tuhinsharma121 committed:
DOC: fix RT03,SA01,ES01 for pandas.core.resample.Resampler.__iter__ (#60033)

KevsterAmp commented:
Oops, merged the wrong branch
